patches/minecraft/net/minecraft/inventory/container/EnchantmentContainer.java.patch

```diff
@@ -76,7 +84,35 @@
                 }
  
                 for(int j1 = 0; j1 < 3; ++j1) {
-@@ -262,9 +251,8 @@
+@@ -168,7 +158,6 @@
+             ItemStack itemstack2 = itemstack;
+             List<EnchantmentData> list = this.func_178148_a(itemstack, p_75140_2_, this.field_75167_g[p_75140_2_]);
+             if (!list.isEmpty()) {
+-               p_75140_1_.func_192024_a(itemstack, i);
+                boolean flag = itemstack.func_77973_b() == Items.field_151122_aG;
+                if (flag) {
+                   itemstack2 = new ItemStack(Items.field_151134_bR);
+@@ -177,12 +166,19 @@
+ 
+                for(int j = 0; j < list.size(); ++j) {
+                   EnchantmentData enchantmentdata = list.get(j);
++                  List<Object> eventOutput = net.minecraftforge.event.ForgeEventFactory.onEnchant(p_75140_1_, itemstack2, enchantmentdata.field_76302_b, enchantmentdata.field_76303_c);
```

Comment:
This fires an event for every enchantment being applied instead of just firing one event with the whole list.
Also that return with a `List<Object>` is _very_ ugly. Just return the event instead?